### ISSUE TITLE
TiDB数据库bit数据类型字段值无法修改为1

### ISSUE THREAD

**Issue by mm20140616** [pingcap/docs-cn#1055]
在执行更新语句时，`bit`数据类型的字段的值无法更新为1。建表的Schema如下：

```text
SET FOREIGN_KEY_CHECKS=0;
DROP TABLE IF EXISTS `DATABASECHANGELOGLOCK`;
CREATE TABLE `DATABASECHANGELOGLOCK` (
  `ID` int(11) NOT NULL,
  `LOCKED` bit(1) NOT NULL,
  `LOCKGRANTED` datetime DEFAULT NULL,
  `LOCKEDBY` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`ID`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci;
INSERT INTO `DATABASECHANGELOGLOCK` VALUES ('1', '', null, null);
```

更新语句如下：

```text
UPDATE DATABASECHANGELOGLOCK SET LOCKED=1 WHERE ID=1 AND LOCKED=0
```

执行`select tidb_version()`获取的数据库版本如下：

```text
Release Version: None
Git Commit Hash: 508f5dd6263e4896d53fb37af99f3a1b747de713
Git Branch: master
UTC Build Time: 2018-10-27 09:59:28
GoVersion: go version go1.11 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
```

**Comment by zyguan** [comment]
Hi, @zhaoxiaoyu19951122 , if you run the following query, you will find that `LOCKED` has been set to 1 actually.
```sql
select * from `DATABASECHANGELOGLOCK` where `LOCKED` = 1
```
[Here](https://asciinema.org/a/d1fpvBWhmSGHBLx4b24BzyVMr) is what I did (note that mysql has the same behavior), do I get you idea correctly?

**Comment by zhaoxiaoyu19951122** [comment]
Hi,@zyguan,the problem I encountered was that the following exception would occur when TiDB set `LOCKED` to 1, causing the update to fail.Here is the exception information
```text
[SQL] UPDATE DATABASECHANGELOGLOCK SET LOCKED=1 WHERE ID=1 AND LOCKED=0
[Err] 2013 - Lost connection to MySQL server during query
```

**Comment by zyguan** [comment]
Ok, I see, the version you use indeed have the problem, however, it seems to be fixed (at least in latest master and release-2.1 branch).
